source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -143,9 +151,31 @@ public Task<TResource> Refresh(TResource resource)
                 if (resource == null) throw new ArgumentNullException("resource");
                 return Get(resource.Id);
             }
-        }
 
-        // ReSharper restore MemberCanBePrivate.Local
+            protected Dictionary<string, object> CreateSpacesParameters(bool includeGlobal, params string[] spaces)
+            {
+                ValidateSpacesParameters(spaces);
+
+                return new Dictionary<string, object>
+                {
+                    ["includeGlobal"] = includeGlobal,
+                    ["spaces"] = spaces
+                };
+            }
+
+            void ValidateSpacesParameters(params string[] spaces)
+            {
+                if (LimitedToSpacesParameters == null) return;
+                var previouslyDefinedSpaceIds = LimitedToSpacesParameters["spaces"] as string[];
+                var previouslyDefinedSpaceIdsSet = new HashSet<string>(previouslyDefinedSpaceIds);
+                if (!previouslyDefinedSpaceIdsSet.IsSupersetOf(spaces))
```

Comment:
This should also take into account `includeGlobal`. If you were previously excluding global, and you have tried to limit it to something that now includesGlobal, this should be an exception.